src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs

```diff
@@ -125,6 +168,34 @@ protected override void ExecuteCore()
                     compositeR2RFileToPublish.RemoveMetadata(MetadataKeys.OriginalItemSpec);
                     compositeR2RFileToPublish.SetMetadata(MetadataKeys.RelativePath, compositeR2RImageRelativePath);
                     r2rFilesPublishList.Add(compositeR2RFileToPublish);
+
+                    string compositePDBImage = null;
+                    string compositePDBRelativePath = null;
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && hasValidDiaSymReaderLib)
+                    {
+                        compositePDBImage = Path.ChangeExtension(compositeR2RImage, ".ni.pdb");
+                        compositePDBRelativePath = Path.ChangeExtension(compositeR2RImageRelativePath, ".ni.pdb");
+                    }
+                    else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                    {
+                        compositePDBImage = Path.ChangeExtension(compositeR2RImage, ".ni.map");
+                        compositePDBRelativePath = Path.ChangeExtension(compositePDBRelativePath, ".ni.map");
```

Comment:
`compositePDBRelativePath` is `null` here. Shouldn't we use `compositeR2RImageRelativePath` instead?

Reply:
For sure, thanks for spotting!

Reply:
Fixed in 15th commit.